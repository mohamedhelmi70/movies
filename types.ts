export type RootParamList = {
    Movies: undefined;
    Movie: {
        movieId: number;
        movie: Movie;
    };
};

export interface Movie {
    poster_path: string;
    adult?: boolean;
    overview: string;
    release_date: string,
    genre_ids: number[],
    id: number,
    original_title: string,
    original_language?: string,
    title: string,
    backdrop_path?: string,
    popularity?: number,
    vote_count?: number,
    video?: boolean,
    vote_average: number,
    genres?: Genre[]
}

export interface Genre {
    id: number;
    name: string;
}

export interface Credit {
    adult?: boolean,
    gender?: number,
    id: number,
    known_for_department?: string,
    name: string,
    original_name: string,
    popularity?: number,
    profile_path: string,
    cast_id?: number,
    character?: string,
    credit_id: string,
    order?: number
}