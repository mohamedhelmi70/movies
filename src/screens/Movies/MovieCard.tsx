import moment from 'moment';
import React from 'react';
import {Image, Pressable, StyleSheet} from 'react-native';
import {Movie} from '../../../types';
import {Text, View} from '../../components/Themed';
import Colors from '../../constants/Colors';
import {wp} from '../../constants/Layout';

interface MovieCardProps {
    movie?: Movie;
    onPress: () => void;
}

function MovieCard({movie, onPress}: MovieCardProps) {
    return (
        <Pressable onPress={onPress} style={styles.card}>
            <View>
                {Boolean(movie?.poster_path) && (
                    <Image
                        source={{uri: `https://www.themoviedb.org/t/p/original${movie?.poster_path}`}}
                        style={styles.poster}
                        resizeMode="cover"
                    />
                )}
            </View>
            <View style={styles.cardRight}>
                <Text style={styles.cardTitle} weight="700" lightColor="#334443">
                    {movie?.original_title}
                </Text>
                <Text style={styles.date} lightColor="#888">
                    {moment(movie?.release_date).format('LL')}
                </Text>
                {movie?.genres && movie?.genres?.length > 0 && (
                    <View style={styles.genresView}>
                        {movie?.genres?.map((genre) => (
                            <View key={genre?.id} style={styles.genre} lightColor="#ccc">
                                <Text style={{fontSize: wp('2.9%')}} weight="400" font="Roboto-Light">
                                    {genre?.name}
                                </Text>
                            </View>
                        ))}
                    </View>
                )}
                {movie?.vote_average && (
                    <View style={{marginTop: wp('2%'), width: wp('60%')}}>
                        <Text style={styles.vote} weight="700" font="Roboto-Medium" lightColor={Colors.light.primary}>
                            {movie?.vote_average * 10}%
                        </Text>
                    </View>
                )}
            </View>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    card: {
        flex: 1,
        width: wp('90%'),
        alignSelf: 'center',
        marginBottom: wp('4%'),
        padding: wp('4%'),
        borderRadius: wp('2%'),
        backgroundColor: '#fff',

        flexDirection: 'row',

        shadowColor: '#888',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.23,
        shadowRadius: 2.62,

        elevation: 4,
    },
    poster: {
        width: wp('18%'),
        height: wp('28%'),
        borderRadius: wp('3%'),
    },
    cardRight: {
        marginLeft: wp('4%'),
    },
    cardTitle: {
        fontSize: wp('4.5%'),
        maxWidth: wp('60%'),
    },
    date: {
        marginTop: wp('2%'),
        fontSize: wp('3.5%'),
    },
    genresView: {
        marginTop: wp('3%'),
        maxWidth: wp('60%'),
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'flex-start',
        flexWrap: 'wrap',
    },
    genre: {
        borderRadius: wp('2%'),
        paddingHorizontal: wp('1.5%'),
        paddingVertical: wp('.5%'),
        marginBottom: wp('1%'),
        marginRight: wp('1%'),
    },
    vote: {
        alignSelf: 'flex-end',
        fontSize: wp('4.3%'),
    },
});

export default MovieCard;
