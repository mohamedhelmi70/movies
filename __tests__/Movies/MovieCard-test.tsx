import React from 'react';
import renderer from 'react-test-renderer';
import {Movie} from '../../types';
import MovieCard from '../../src/screens/Movies/MovieCard';

test('renders MovieCard correctly', () => {
    const mockCallBack = jest.fn();
    const movieItem: Movie = {
        id: 123,
        original_title: 'test',
        poster_path: '/poster.png',
        vote_average: 3,
        title: 'test',
        genre_ids: [1, 2],
        release_date: '1994-09-10',
        overview: 'overview test',
    };
    const tree = renderer.create(<MovieCard movie={movieItem} onPress={mockCallBack} />).toJSON();
    expect(tree).toMatchSnapshot();
});
