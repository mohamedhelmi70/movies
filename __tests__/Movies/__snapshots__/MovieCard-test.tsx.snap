// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieCard correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "borderRadius": 15,
      "elevation": 4,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 30,
      "padding": 30,
      "shadowColor": "#888",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
      "width": 675,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 22.5,
            "height": 210,
            "width": 135,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "cache": "immutable",
            "headers": Object {
              "Authorization": "4f298a53e552283bee957836a529baec",
            },
            "priority": "normal",
            "uri": "https://www.themoviedb.org/t/p/original/poster.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        Object {
          "marginLeft": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#334443",
            "fontFamily": "Roboto",
            "fontWeight": "700",
            "textAlign": "left",
          },
          Object {
            "fontSize": 34,
            "maxWidth": 450,
          },
        ]
      }
    >
      test
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#888",
            "fontFamily": "Roboto",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Object {
            "fontSize": 26.5,
            "marginTop": 15,
          },
        ]
      }
    >
      September 10, 1994
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
          },
          Object {
            "marginTop": 15,
            "width": 450,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#79c700",
              "fontFamily": "Roboto",
              "fontWeight": "700",
              "textAlign": "left",
            },
            Object {
              "alignSelf": "flex-end",
              "fontSize": 32.5,
            },
          ]
        }
      >
        30
        %
      </Text>
    </View>
  </View>
</View>
`;
